packages/gatsby/src/services/initialize.ts

```diff
@@ -336,25 +328,30 @@ module.exports = async (args: BootstrapArgs) => {
     return undefined
   }
 
-  const ssrPlugins = _.filter(
-    flattenedPlugins.map(plugin => {
+  interface IPluginResolution {
+    resolve: string
+    options: IPluginInfoOptions
+  }
+
+  const isResolved = (plugin): plugin is IPluginResolution => !!plugin.resolve
```

Comment:
TIL about `is`. Seems valid. Not quite convinced that's a pattern we should use.

Reply:
It's super useful, because it narrows the type when filtering an array so you don't need to do extra typechecking.